ion-schema/src/constraint.rs

```diff
@@ -71,32 +72,58 @@ pub enum Constraint {
 impl Constraint {
     /// Creates a [Constraint::Type] referring to the type represented by the provided [TypeId].
     pub fn type_constraint(type_id: TypeId) -> Constraint {
-        Constraint::Type(TypeConstraint::new(type_id))
+        Constraint::Type(TypeConstraint::new(TypeReference::new(
```

Comment:
All the changes in this file are due to storing `TypeReference`s instead of `TypeId`s for all the constraints